core/spec/models/spree/address_spec.rb

```diff
@@ -119,6 +119,47 @@
     end
   end
 
+  context "when updating a record" do
+    context "if use_combined_first_and_last_name_in_address is set to false (default)" do
+      before do
+        allow(Spree::Config).to receive(:use_combined_first_and_last_name_in_address) { false }
+      end
+
+      context "and the `name` attribute is not in changeset" do
+        let(:address) { create(:address, firstname: 'John', lastname: 'Doe') }
+
+        it "sets `name` from `firstname` and `lastname`" do
+          new_address = described_class.immutable_merge(address, firstname: 'Jane', lastname: 'Fonda')
+          expect(new_address.read_attribute(:name)).to eq('Jane Fonda')
+        end
+      end
+    end
+
+    context "if use_combined_first_and_last_name_in_address is set to true" do
+      before do
+        allow(Spree::Config).to receive(:use_combined_first_and_last_name_in_address) { true }
+      end
+
+      context "and the `name` attribute is not in changeset" do
+        let(:address) { create(:address, firstname: 'John', lastname: 'Doe') }
+
+        it "keeps old name" do
+          new_address = described_class.immutable_merge(address, firstname: 'Jane', lastname: 'Fonda')
+          expect(new_address.read_attribute(:name)).to eq('John Von Doe')
+        end
+      end
+
+      context "and the `name` attribute is in changeset" do
+        let(:address) { create(:address, firstname: 'John', lastname: 'Doe') }
+
+        it "keeps old name" do
```

Comment:
Reading the test expectation, it doesn't seem the old name is kept, or am I missing something?

Reply:
Yes, you are right. Fixing

Reply:
fixed